llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -870,6 +870,11 @@ void MachineCopyPropagation::forwardUses(MachineInstr &MI) {
     ++NumCopyForwards;
     Changed = true;
   }
+  // Attempt to canonicalize/optimize the instruction now its arguments have
+  // been mutated.
+  if (TII->optimizeInstruction(MI)) {
+    LLVM_DEBUG(dbgs() << "MCP: After optimizeInstruction: " << MI);
```

Comment:
Do we need to set `Changed = true` here or is guaranteed it was already set earlier?

Reply:
Good question. It's not clear to me that Changed = true will _always_ have been set, so I've added an explicit assignment so it's obvious.